lib/auth/middleware.go

```diff
@@ -189,7 +189,9 @@ func NewTLSServer(cfg TLSServerConfig) (*TLSServer, error) {
 		cfg: cfg,
 		httpServer: &http.Server{
 			Handler:           tracingHandler,
-			ReadHeaderTimeout: apidefaults.DefaultIOTimeout,
+			ReadTimeout:       apidefaults.DefaultIOTimeout,
+			ReadHeaderTimeout: defaults.ReadHeadersTimeout,
```

Comment:
> What do you think about starting with 2 seconds, and leaving this in master for the v14 performance tests before backporting?

Sounds like a great plan to me.  In commit ff22401ecd6b5c95793e8c03cba3ea77e1c642e2 I updated the configured timeout to 10 seconds.  I feel like 2 seconds is likely fine, but we are coming from 30 seconds so even at 10 I feel like this is a notable net gain.

Maybe in Teleport 15 we go to 2 seconds (no concerns with moving slow to make sure we don't cause impacts)

Reply:
I have seen two test failures, I worry this could be increasing the flakiness.

The timeouts seem pretty reasonable though, does it make sense to reduce test concurrency so that tests can complete faster?  @zmb3 what are your thoughts?

Reply:
Which tests? It's probably Go 1.21 and not your change

Reply:
Seen twice:
```
 === Failed
=== FAIL: lib/auth TestAutoRotation (1.75s)
    tls_test.go:418: 
        	Error Trace:	/__w/teleport/teleport/lib/auth/tls_test.go:418
        	Error:      	Error "write tcp 127.0.0.1:47596->127.0.0.1:32991: write: broken pipe" does not contain "certificate"
        	Test:       	TestAutoRotation
```

I assumed it might be the write timeout causing the connection to be closed.

Reply:
https://github.com/gravitational/teleport/issues/30253
may or may not be your change but i don't think we have found the root cause for the above flaky test yet